src/trupti/Shop.java

```diff
@@ -0,0 +1,109 @@
+package trupti;
+/*Assignment  7  : 30th March 2021 
+1. Create a system (Shop) which maintains each item's quantity.
+a. Initial stocks of Maggie: 50 Maggie packets 
+b. Initial stocks of Dosa : 43 packets 
+c. Initial stocks of pouches : 39 oil packets 
+d. Initial stocks of panipuri : 43 packets  
+e. Initial stocks of masala�s : 73 packets.*/
+
+public class Shop {
+	String maggie,dosa,pouches,panipuri,masala;
+	int maggieStock,dosaStock,pouchesStock,panipuriStock,masalaStock;
+	
+	void setInitialQuantity(int maggie,int dosa,int pouches,int panipuri,int masala) {
+		maggieStock=maggie;
+		dosaStock=dosa;
+		pouchesStock=pouches;
+		panipuriStock=panipuri;
+		masalaStock=masala;
+	}
+	
+	void requestedQuantity(int reqMaggie,int reqDosa,int reqPouches,int reqPanipuri,int reqMasala,String pAllowedStatus) {
+		if(reqMaggie<=maggieStock) {
+			maggieStock=maggieStock-reqMaggie;
+		}
+		else {
+			System.out.println("Sorry,not sufficient stock.We can give you only "+maggieStock);
+			System.out.println("Maggie in stock is : "+maggieStock+" and requested quantity is :"+reqMaggie+" Running out of stock ");
+		if(pAllowedStatus.equals("PAllowed")) {
+			maggieStock=0;
+		}
+		}
+		
+		if(reqDosa<=dosaStock) {
+			dosaStock=dosaStock-reqDosa;
+		}
+		else {
+			System.out.println("Sorry,not sufficient stock.We can give you only "+dosaStock);
+			System.out.println("Dosa in stock is : "+dosaStock+" and requested quantity is :"+reqDosa+" Running out of stock ");
+		if(pAllowedStatus.equals("PAllowed")) {
+			dosaStock=0;
```

Comment:
If request stock is not available mean we are setting direct it zero, 
what if second time your calling same method and let say you request 20 packs, then this looks not logically correct